### ISSUE TITLE
Feature-Request: Shutdown

### ISSUE THREAD

**Issue by derDieDasJojo** [guardianproject/haven#200]
I have the following Idea:
Add the option to shut down the phone if the configured sensors show changes.
This could bring additional physical safety to services running on the haven-phone.
For example the phone could run a fileserver, a webserver,..
and if any physical action around is happening it turns into a secure state and shuts down.
if the device is encrypted the data is then pretty save from any attacker.
Only the authorized user knows the password and can start the phone, decrypt the device and
(re)configure the service

**Comment by petrk94** [comment]
The idea is not bad +1

**Comment by n8fr8** [comment]
Agreed.. .a slightly different use case, but definitely useful for that need.

**Comment by percy-g2** [comment]
just executing reboot -p with Runtime.getRuntime().exec() will do on even trigger will do.

**Comment by fat-tire** [comment]
This is an interesting idea-- the use case would be different from how haven has been thought of previously, which is as an add-on device placed on top of the "real" valuable object (laptop, say).  In this scenario, the haven device itself is running some vital services, and physically disturbing it would trigger an automatic shutdown & lockdown, yah?

I could see how this could be useful.

From this stackoverflow [article](https://stackoverflow.com/questions/32984849/restart-android-device-programmatically), it looks like the required permission to do a shutdown is `android.permission.REBOOT`, which may not be assigned to a regular app, only one signed with the system firmware signing key (or w/root access obviously).

Also see [here](https://developer.android.com/reference/android/Manifest.permission.html#REBOOT).

**Comment by fat-tire** [comment]
How about-- since we're talking about potentially rooting this device or otherwise making it a system app-- how about having it run an entire user-defined "panic script" as root-- a bash script running in a root shell.  It could do whatever you want-- `setenforce 0`, then wipe files, kill servers, format partitions, launch apps, shut down or whatever.  And/or could even have a little wizard type of thing using that [appintro](https://github.com/apl-devs/AppIntro) library to "build" a list of things you want to have happen.

**Comment by leonardoporpora** [comment]
I would add shutdown when the usb plugin is triggered because it could be connected to a pc and someone could delete files and other stuff
